tests/emterpreter_async_bad_2.cpp

```diff
@@ -11,12 +11,9 @@ extern "C" {
 
 void EMSCRIPTEN_KEEPALIVE finish(int result) {
   EM_ASM({
-    var xhr = new XMLHttpRequest();
-    xhr.open("GET", "http://localhost:8888/report_result?1");
-    xhr.onload = xhr.onerror = function() {
-      window.close();
-    };
-    xhr.send();
+    if (Module.reported) return;
```

Comment:
if `reportResultToServer` checks if we've reported already, why do it with Module too?

Reply:
This is specific to this test - `finish` may be called more than once, and we care about the first call. In other words, it's not an error in the test to call that function more than once, but it would be an error to send the result more than once.

Reply:
Isn't that what maybeReportResultToServer is for?